midenc-driver/src/commands/compile/stages/rewrite.rs

```diff
@@ -26,10 +26,12 @@ impl Stage for ApplyRewritesStage {
         let matches = session.matches();
         for rewrite in inventory::iter::<RewritePassRegistration<hir::Module>> {
             let flag = rewrite.name();
-            if let Some(index) = matches.index_of(flag) {
-                let is_enabled = matches.get_flag(flag);
-                if is_enabled {
-                    registered.push((index, rewrite.get()));
+            if matches.try_contains_id(flag).is_ok() {
```

Comment:
Otherwise, `ApplyRewritesStage` crashes on `matches.get_flag` when arg matches are not provided in `Session`. The same applies to the second fix below. I made these fixes, assuming that `Session.matches` is not required. Let me know if I'm missing something.